### DISCUSSION TITLE
[question] why do we "need" a `defaultTheme`?

### DISCUSSION THREAD

**Discussion by FranciscoKloganB** [RyanClementsHax/tailwindcss-themer#91] [Q&A]
So, this is not my first time using this plugin, but now I was left wondering...

Why does the plugin require a `defaultTheme` and `themes` instead of just `themes`?

Isn't the `defaultTheme` the exact samething as using Tailwind's own `theme`? If not, how are these merged together?

From my point of view we have 3 ways of defining and extending our tailwind APIs, when we should have two? I am sure I am missing something and that's my goal. To understand why we have `theme`, `defaultTheme` and `themes`.

```js
const preset = {
  attributify: false,
  darkMode: "class",
  theme: {
     // Override tailwind default theme configuration
     ...tailwindThemeOverride
     extend: {
       // Add new things to tailwind default theme configuration
       ...tailwindThemeExtends
     }
  },
  plugins: [
    require("tailwindcss-themer")({
      defaultTheme: {
        extend: {
          // 👇👇👇👇
          // Question: How does this differ from `theme` or `theme.extend`? How are they merged?
        },
      },
      themes: [
        {
          name: "light",
          extend: {
             // 👇👇👇👇
             // Question: Does this override `defaultTheme`, or `theme.extend`, or `theme`?
           }
        },
        {
          name: "dark",
        },
      ],
    }),
  ],
}
```

**Comment by RyanClementsHax** [comment]
Thanks for your question! I'd be happy to give an answer.

TLDR: a limitation of tailwind's plugin api. Upvote https://github.com/tailwindlabs/tailwindcss/discussions/6980 if you want to see this go away.

You're actually onto something. I agree that it would be cleaner to not even have `defaultTheme` and rather specify all of those values in the normal tailwind config. Unfortunately, tailwind plugins don't have a way of reading default tailwind configuration when creating your own overrides for theming. The nitty gritty details are all in https://github.com/tailwindlabs/tailwindcss/discussions/6980 if you want to learn more.

Does this answer your question?

(Also thanks for your patience, things have been crazy on my end and now have gotten better where I can get back to open source)
